plugins/nanospell/plugin.js

```diff
@@ -664,19 +664,17 @@
 				var blockList = event.data.blockList;
 				var url = resolveAjaxHandler();
 				var callback = function (data) {
-					var bookmarks;
+					var selectionStart = editor.getSelection().getStartElement();
+					var rootElement;
 					parseRpc(data, words);
-					editor.lockSelection();
-					bookmarks = editor.getSelection().createBookmarks(true);
+
 					for (var i = 0; i < blockList.length; i++) {
-						var rootElement = blockList[i];
+						rootElement = blockList[i];
 						editor.fire(EVENT_NAMES.START_RENDER, {
 							root: rootElement,
-							needsBookmarkCreated: false,
+							needsBookmarkCreated: selectionStart ? rootElement.contains(selectionStart) || rootElement.equals(selectionStart) : null,
```

Comment:
It's a wrapper for the either `compareDocumentPosition` (https://developer.mozilla.org/en-US/docs/Web/API/Node/compareDocumentPosition) `Node.contains` (https://developer.mozilla.org/en/docs/Web/API/Node/contains).


Intuitively your point makes sense, but in reality, in my lorem ipsum paste document, in Chrome, `contains` usually around 0.1ms, with some odd calls to it taking over 1ms.  I think we can leave this alone.

Also, once we make render called async, we can't actually move this out of the "loop".  You can move the cursor into a block before the timeout fires, if the UI is more responsive, for example.